### PR TITLE
Add new metrics for DR and Perf replication

### PR DESCRIPTION
**Pull request by seanmalloy**
Perf and DR replication are Vault Enterprise
features. These metrics will always be "0" for
open source vault. Below are the list of new
metrics being added.

* vault_replication_dr_primary
* vault_replication_dr_secondary
* vault_replication_performance_primary
* vault_replication_performance_secondary

### CODE REVIEW

**Comment by seanmalloy** [comment]
@jdbaldry any details on why the Jenkins CI build failed?

**Comment by jdbaldry** [comment]
Hey! Thanks for the PR. Unfortunately changes in the org mean that we cannot expose a read only jenkins any more. There's a linting error thrown on the snake case variables. I'll add a review.

@tomwilkie Would you be interested in this being moved to your Org for the future? There's a whole process involved in keeping this up now we've been acquired.

File: main.go [hunk 1]
```diff
@@ -55,6 +55,26 @@ var (
 		"Is this Vault node in standby.",
 		nil, nil,
 	)
+	replication_dr_primary = prometheus.NewDesc(
```
Comment (jdbaldry) [inline main.go]:
```suggestion
	replicationDRPrimary = prometheus.NewDesc(
```

File: main.go [hunk 2]
```diff
@@ -55,6 +55,26 @@ var (
...
+		prometheus.BuildFQName(namespace, "", "replication_dr_primary"),
+		"Is this Vault node a primary disaster recovery replica.",
+		nil, nil,
+	)
+	replication_dr_secondary = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "", "replication_dr_secondary"),
+		"Is this Vault node a secondary disaster recovery replica.",
+		nil, nil,
+	)
+	replication_performance_primary = prometheus.NewDesc(
```
Comment (jdbaldry) [inline main.go]:
```suggestion
	replicationPerformancePrimary = prometheus.NewDesc(
```

File: main.go [hunk 3]
```diff
@@ -55,6 +55,26 @@ var (
 		"Is this Vault node in standby.",
 		nil, nil,
 	)
+	replication_dr_primary = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "", "replication_dr_primary"),
+		"Is this Vault node a primary disaster recovery replica.",
+		nil, nil,
+	)
+	replication_dr_secondary = prometheus.NewDesc(
```
Comment (jdbaldry) [inline main.go]: replicationDRSecondary

File: main.go [hunk 4]
```diff
@@ -55,6 +55,26 @@ var (
...
+		prometheus.BuildFQName(namespace, "", "replication_dr_secondary"),
+		"Is this Vault node a secondary disaster recovery replica.",
+		nil, nil,
+	)
+	replication_performance_primary = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "", "replication_performance_primary"),
+		"Is this Vault node a primary performance replica.",
+		nil, nil,
+	)
+	replication_performance_secondary = prometheus.NewDesc(
```
Comment (jdbaldry) [inline main.go]:
```suggestion
	replicationPerformanceSecondary = prometheus.NewDesc(
```

**Comment by josephschadlick** [comment]
Hello, I am excited about this PR! Is there anything blocking this aside from the syntax changes?

**Comment by seanmalloy** [comment]
I will take a look at this in the next week, and get the fmt/lint issues fixed.

**Comment by jdbaldry** [comment]
Looks good to me, thanks for the contribution!

**Comment by jdbaldry** [comment]
Merged and released as 0.1.2, thanks again

**Pull request merged**